spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+RSpec.describe 'Customer Show Page', type: :feature do
+  describe 'As a visitor' do
+    before(:each) do
+      @supermarket_1 = Supermarket.create!(name: "Ralphs", location: "Over There St.")
+      @customer_1 = Customer.create!(name: "Alex Long")
+      @item_1 = @supermarket_1.items.create!(name: "Apple", price: 5)
+      @item_2 = @supermarket_1.items.create!(name: "Banana", price: 2)
+      @item_3 = @supermarket_1.items.create!(name: "Orange", price: 3)
+      @item_4 = @supermarket_1.items.create!(name: "Onion", price: 1)
+      @item_5 = @supermarket_1.items.create!(name: "Bell Pepper", price: 4)
+      @customer_item_1 = CustomerItem.create!(customer_id: @customer_1.id, item_id: @item_1.id)
+      @customer_item_2 = CustomerItem.create!(customer_id: @customer_1.id, item_id: @item_2.id)
+      @customer_item_3 = CustomerItem.create!(customer_id: @customer_1.id, item_id: @item_3.id)
+    end
+
+    it 'shows the customers name and list of its items' do
+      visit "/customers/#{@customer_1.id}"
+
+      expect(page).to have_content(@customer_1.name)
+      expect(page).to have_content(@item_1.name)
+      expect(page).to have_content(@item_2.name)
+      expect(page).to have_content(@item_3.name)
+    end
+
+    it 'shows the price of each item and the supermarket it belongs to' do
+      visit "/customers/#{@customer_1.id}"
+
+      within "#item-#{@item_1.id}" do
+        expect(page).to have_content(@item_1.price)
+        expect(page).to have_content(@supermarket_1.name)
+      end
+      within "#item-#{@item_2.id}" do
+        expect(page).to have_content(@item_2.price)
+        expect(page).to have_content(@supermarket_1.name)
+      end
+      within "#item-#{@item_3.id}" do
+        expect(page).to have_content(@item_3.price)
+        expect(page).to have_content(@supermarket_1.name)
+      end
+    end
```

Comment:
If I gave you this code and told me what the `5` was supposed to represent, I would venture a guess that you probably wouldn't be able to:

```ruby
within "#item-#{@item_1.id}" do
  expect(page).to have_content(5)
  expect(page).to have_content("Ralphs")
end
```

How about now:

```ruby
within "#item-#{@item_1.id}" do
  expect(page).to have_content("Price: 5")
  expect(page).to have_content("Supermarket: Ralphs")
end
```

That's not to say that you have to write everything out, but just making use of some "anchor" text makes a world of difference:

```ruby
within "#item-#{@item_1.id}" do
  expect(page).to have_content("Price: #{@item_1.price}")
  expect(page).to have_content("Supermarket: #{@supermarket_1.name}")
end
```

While within blocks do in fact reduce some ambiguity, we can take it further with anchor text as well.